examples/raw-example.js

```diff
@@ -1,20 +1,49 @@
-const k8s = require('@kubernetes/client-node');
-const request = require('request');
+// in a real program use require('@kubernetes/client-node')
+const k8s = require('../dist/index');
+const fetch = require('node-fetch');
+const https = require('https');
+const yaml = require('js-yaml');
+const fs = require('fs');
 
 const kc = new k8s.KubeConfig();
 kc.loadFromDefault();
 
-const opts = {};
-kc.applyToRequest(opts);
+const KUBECONFIG = process.env.KUBECONFIG;
+if (!KUBECONFIG) {
+    throw Error('please give me a KUBECONFIG path to work with.');
+}
 
-request.get(`${kc.getCurrentCluster().server}/api/v1/namespaces/default/pods`, opts,
-    (error, response, body) => {
-        if (error) {
-            console.log(`error: ${error}`);
-        }
-        if (response) {
-            console.log(`statusCode: ${response.statusCode}`);
-        }
-        console.log(`body: ${body}`);
-  });
+const kubeconfig = yaml.load(fs.readFileSync(KUBECONFIG, 'utf-8'));
+console.log(kubeconfig);
+if (!kubeconfig || !kubeconfig.users || !kubeconfig.clusters || !kubeconfig['current-context']) {
+    throw Error('please give me a valid KUBECONFIG file to work with.');
+}
+const cluster = kubeconfig.clusters.find((c) => c.name === kubeconfig['current-context']).cluster;
+const user = kubeconfig.users.find((u) => u.name === kubeconfig['current-context']).user;
+
+const agent = new https.Agent({
```

Comment:
If you have a better way to authenticate then LMK!

Reply:
Did you try using `applyToHTTPSOptions` or `applyToFetchOptions`? 
I'm not sure it will work but I guess this should be the equivalent of `applyToRequest`.

* https://github.com/kubernetes-client/javascript/blob/release-1.x/src/config.ts#L150
* https://github.com/kubernetes-client/javascript/blob/release-1.x/src/config.ts#L173

Reply:
I did, but unsuccessfully. I might go back and try again. It wanted some context that I could get from the library. Not saying it can't be done though...

Reply:
I figured it out! Thanks for the comment! Made me go back and try again 🥇